pilot/pkg/model/xds_cache.go

```diff
@@ -104,329 +47,218 @@ type XdsCache interface {
 	Clear(sets.Set[ConfigKey])
 	// ClearAll clears the entire cache.
 	ClearAll()
-	// Keys returns all currently configured keys. This is for testing/debug only
-	Keys() []string
-	// Snapshot returns a snapshot of all keys and values. This is for testing/debug only
-	Snapshot() map[string]*discovery.Resource
-}
-
-// NewXdsCache returns an instance of a cache.
-func NewXdsCache() XdsCache {
-	cache := &lruCache{
-		enableAssertions: features.EnableUnsafeAssertions,
-		configIndex:      map[ConfigHash]sets.String{},
-		evictQueue:       make([]evictKeyConfigs, 0, 1000),
-	}
-	cache.store = newLru(cache.onEvict)
-	return cache
-}
-
-type evictKeyConfigs struct {
-	key              string
-	dependentConfigs []ConfigHash
+	// Keys returns all currently configured keys for the type. This is for testing/debug only
+	Keys(t string) []any
+	// Snapshot returns a snapshot of all values. This is for testing/debug only
+	Snapshot() []*discovery.Resource
 }
 
-type lruCache struct {
-	enableAssertions bool
-	store            simplelru.LRUCache[string, cacheValue]
-	// token stores the latest token of the store, used to prevent stale data overwrite.
-	// It is refreshed when Clear or ClearAll are called
-	token       CacheToken
-	mu          sync.RWMutex
-	configIndex map[ConfigHash]sets.String
-
-	evictQueue []evictKeyConfigs
-
-	// mark whether a key is evicted on Clear call, passively.
-	evictedOnClear bool
-}
-
-var _ XdsCache = &lruCache{}
-
-func newLru(evictCallback simplelru.EvictCallback[string, cacheValue]) simplelru.LRUCache[string, cacheValue] {
-	sz := features.XDSCacheMaxSize
-	if sz <= 0 {
-		sz = 20000
-	}
-	l, err := simplelru.NewLRU(sz, evictCallback)
-	if err != nil {
-		panic(fmt.Errorf("invalid lru configuration: %v", err))
-	}
-	return l
+// XdsCacheEntry interface defines functions that should be implemented by
+// resources that can be cached.
+type XdsCacheEntry interface {
+	Type() string
+	// Key is the key to be used in cache.
+	Key() any
+	// DependentConfigs is config items that this cache key is dependent on.
+	// Whenever these configs change, we should invalidate this cache entry.
+	DependentConfigs() []ConfigHash
+	// Cacheable indicates whether this entry is valid for cache. For example
+	// for EDS to be cacheable, the Endpoint should have corresponding service.
+	Cacheable() bool
 }
 
-func (l *lruCache) Run(stop <-chan struct{}) {
-	interval := features.XDSCacheIndexClearInterval
-	go func() {
-		ticker := time.NewTicker(interval)
-		defer ticker.Stop()
-		for {
-			select {
-			case <-ticker.C:
-				l.mu.Lock()
-				for _, keyConfigs := range l.evictQueue {
-					l.clearConfigIndex(keyConfigs.key, keyConfigs.dependentConfigs)
-				}
-				l.evictQueue = l.evictQueue[:0:1000]
-				l.recordDependentConfigSize()
-				l.mu.Unlock()
-			case <-stop:
-				return
-			}
-		}
-	}()
-}
+const (
```

Comment:
May be we should use typeUrls for safety https://github.com/istio/istio/blob/master/pilot/pkg/xds/v3/model.go#L26

Reply:
it is just a flag, not being stored at all

Reply:
I know. I am just suggesting use the same const